File: R/getbasis.R
getbasis <- function(fd) {
  #  Extracts the basis.fd object from a functional data object FD, or,
  #    if FD is already a basis object, just returns it.

  #  Last modified 4 July 2001
  
  if (inherits(fd, "fd")) {
    basisfd <- fd[[2]]
  }  else {
    if (inherits(fd, "basis.fd")) {
      basisfd <- fd
    } else {
      stop("An object of class fd or basis.fd expected, but not found.")
    }
  }
  return(basisfd)
}

File: inst/Matlab/fdaM/@fd/getbasis.m
function basisobj = getbasis(fd)
%  GETBASIS    Extracts basis.fd object from functional data object.
%     FD, or, if FD is already a basis object, just returns it.

%  last modified 20 July 2006

  if isa_fd(fd) || isa_basis(fd)
    if isa_fd(fd)
      basisobj = fd.basisobj;
    else
      basisobj = fd;
    end
  else
    error('Structure is neither of fd type or of basis type');
  end

